Intersect.Client/Entities/Player.cs

```diff
@@ -742,37 +742,78 @@ public void TryDepositItem(int inventorySlotIndex, int bankSlotIndex = -1)
-                {
-                    ChatboxMsg.AddMessage(new ChatboxMsg(Strings.Guilds.NotAllowedDeposit.ToString(Globals.Me.Guild), CustomColors.Alerts.Error, ChatMessageType.Bank));
-                    return;
-                }
+                ChatboxMsg.AddMessage(new ChatboxMsg(Strings.Guilds.NotAllowedDeposit.ToString(Globals.Me.Guild), CustomColors.Alerts.Error, ChatMessageType.Bank));
+                return;
             }
 
-            var inventoryQuantity = GetQuantityOfItemInInventory(itemDescriptor.Id);
-            if (inventorySlot.Quantity < 2)
+            var itemQuantityInInventory = GetQuantityOfItemInInventory(itemDescriptor.Id);
+            var availableSpace = FindAvailableBankSpaceForItem(inventorySlot.ItemId, itemQuantityInInventory);
+
+            if (availableSpace < 1)
             {
-                PacketSender.SendDepositItem(inventorySlotIndex, 1, bankSlotIndex);
+                ChatboxMsg.AddMessage(new ChatboxMsg(Strings.Bank.NoSpace, CustomColors.Alerts.Error,
+                    ChatMessageType.Bank));
+                return;
             }
 
-            var userData = new int[2] { inventorySlotIndex, bankSlotIndex };
+            if (itemQuantityInInventory < 2)
+            {
+                DepositSingleItem(inventorySlotIndex, bankSlotIndex);
+                return;
+            }
+
+            OpenDepositItemInputBox(itemDescriptor, inventorySlotIndex, bankSlotIndex, availableSpace);
+        }
+
+        private bool IsGuildBankDepositAllowed()
+        {
+            var rank = Globals.Me.GuildRank;
+            return !string.IsNullOrWhiteSpace(Globals.Me.Guild) && (rank.Permissions.BankDeposit || Globals.Me.Rank == 0);
+        }
+
+        private void DepositSingleItem(int inventorySlotIndex, int bankSlotIndex)
+        {
+            PacketSender.SendDepositItem(inventorySlotIndex, 1, bankSlotIndex);
+        }
 
+        private void OpenDepositItemInputBox(ItemBase itemDescriptor, int inventorySlotIndex, int bankSlotIndex, int availableSpace)
+        {
             InputBox.Open(
                 title: Strings.Bank.deposititem,
                 prompt: Strings.Bank.deposititemprompt.ToString(itemDescriptor.Name),
                 modal: true,
                 inputType: InputBox.InputType.NumericSliderInput,
                 onSuccess: DepositItemInputBoxOkay,
                 onCancel: null,
-                userData: userData,
-                quantity: inventorySlot.Quantity,
-                maxQuantity: inventoryQuantity
-            );
+                userData: new[] { inventorySlotIndex, bankSlotIndex },
+                quantity: availableSpace,
+                maxQuantity: availableSpace);
+        }
+
+        private static int FindAvailableBankSpaceForItem(Guid itemId, int itemQuantityInInventory)
+        {
+            int spaceLeft = 0;
+            int maxBankStack = ItemBase.Get(itemId).MaxBankStack;
```

Comment:
There's no null safety here. If the item descriptor is not retrievable, return 0 immediately, even before allocating `spaceLeft`.

i.e.

```csharp
if (!ItemBase.TryGet(itemId, out var itemDescriptor))
{
    return 0;
}
var maxBankStack = itemDescriptor.MaxBankStack;
var spaceLeft = 0;
```